T2- Valores nulos (Maria)

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns

df = sns.load_dataset("titanic")

Carga del dataset

In [2]:
df.isna().sum()

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

Detectar valores nulos

In [3]:
(df.isna().sum() / len(df)) * 100

survived        0.000000
pclass          0.000000
sex             0.000000
age            19.865320
sibsp           0.000000
parch           0.000000
fare            0.000000
embarked        0.224467
class           0.000000
who             0.000000
adult_male      0.000000
deck           77.216611
embark_town     0.224467
alive           0.000000
alone           0.000000
dtype: float64

Util. Permite saber qué columnas tienen problemas importantes

In [4]:
df["age"] = df["age"].fillna(df["age"].median())

Limpieza de edad. La edad tiene algunos valores extremos y la mediana es mejor para ellos. Además, es una variable numérica

In [5]:
df["embarked"] = df["embarked"].fillna(df["embarked"].mode()[0])

embarked es una variable categórica. No podemos ponerle una media. La moda es el valor que aparece más veces

In [6]:
df["embark_town"] = df["embark_town"].fillna(df["embark_town"].mode()[0])

embark_town tiene 2 valores nulos y es también categórica.

In [7]:
df = df.drop(columns=["deck"])

La columna deck presenta 688 valores nulos de un total de 891 registros, aproximadamente el 77,2 %. Debido a que la mayoría de los datos de esta variable están ausentes, se ha decidido eliminar la columna en lugar de imputar sus valores, ya que una imputación tan extensa introduce información artificial.

In [8]:
df.isna().sum()

survived       0
pclass         0
sex            0
age            0
sibsp          0
parch          0
fare           0
embarked       0
class          0
who            0
adult_male     0
embark_town    0
alive          0
alone          0
dtype: int64

Comprobación de limpieza

In [10]:
df.to_csv("titanic_limpio.csv", index=False)

Guardo el nuevo contenido en un archivo csv para el resto de compañeros